packages/filesystem/src/browser/file-tree/file-tree.ts

```diff
@@ -78,6 +90,38 @@ export class FileTree extends TreeImpl {
         };
     }
 
+    protected isVirtualRoot(node: CompositeTreeNode): boolean {
+        return node.parent === undefined && this.hasVirtualRoot && node.name === 'WorkspaceRoot';
```

Comment:
Is it possible to simply test the node based on a field it has ?
Like instead of checking if the tree knows if it holds a virtual root and check the name, have a field `virtual` and test like `!node.parent && node.virtual`. Or even `!node.parent && VirtualNode.is(node)` to match the way the check is done on other nodes ?

Maybe the tree doesn't have to know if it holds a virtual node, that's my thought. Tell me if I missed something.

Reply:
I believe I don't event need the flag `hasVirtualRoot` in `FileTree` class. Like you said, the tree shouldn't have the knowledge pertaining to its root is virtual or not.